status/condition_set.go

```diff
@@ -237,3 +239,60 @@ func (c ConditionSet) findUnhealthyDependents() []Condition {
 	})
 	return conditions
 }
+
+// objectImpl is an embedded struct type that embeds the upstream type
```

Comment:
Can we document that this assumes that you have your status conditions sitting on the `status.conditions` path?  I think this is a common enough paradigm that we can pretty much assume that conditions will _always_ sit on this path, but the adapter that we are using here (as well as the `FromUnstructured` function) should document that this is the case